### PR TITLE
key_map.rs: propagate "invalid key" errors from multipath keys

### PR DESCRIPTION
**Pull request by apoelstra**
Since #851 we return a `NotSupported` error when attempting to request an xpriv from an x-only key. I'm not sure if this is the correct semantics, but it was deliberately written, so we'll keep it. **However** the behavior was different for multipath xprivs. There we returned `None`, i.e. "key not found/mismatch" rather than an error.

This PR returns an error in both cases.

### CODE REVIEW

**Review by tcharding — APPROVED** [review]
ACK df8d86011d51731c7d0d8af97263b90db22c2384

File: src/descriptor/key_map.rs [hunk 1]
```diff
@@ -336,4 +336,81 @@ mod tests {
...
+        let sk =
+            PrivateKey::from_str("cMk8gWmj1KpjdYnAWwsEDekodMYhbyYBhG8gMtCCxucJ98JzcNij").unwrap();
+        let xonly_pk = sk.public_key(&secp).inner.x_only_public_key().0;
+        let request_x = KeyRequest::XOnlyPubkey(xonly_pk);
+
+        let result = descriptor_sk.get_key(request_x.clone(), &secp);
+        assert!(matches!(result, Err(GetKeyError::NotSupported)));
+
+        // Also test with KeyMap; as in the previous test, the error turns to None.
+        let descriptor_s = "wpkh([d34db33f/84h/1h/0h]tprv8ZgxMBicQKsPd3EupYiPRhaMooHKUHJxNsTfYuScep13go8QFfHdtkG9nRkFGb7busX4isf6X9dURGCoKgitaApQ6MupRhZMcELAxTBRJgS/*)";
```
Comment (oleonardolima) [inline src/descriptor/key_map.rs:408]: question: any reason not to test with the multipath here too?

**Review by oleonardolima — COMMENTED** [review]
Thanks! Sorry for the oversight on the multipath descriptors.

Overall it looks good to me, just a minor comment on the test. I'll apply the tests to the backport PR.

**Review by apoelstra — COMMENTED** [review]
On 89b5048c49c905fa3717200081aa75011ad43599 successfully ran local tests

**Review by oleonardolima — COMMENTED** [review]
ACK 89b5048c49c905fa3717200081aa75011ad43599

File: src/descriptor/key_map.rs (hunk 1 above)
Comment (oleonardolima) [reply src/descriptor/key_map.rs]: Although I suggested the changes, I noticed that I wont' be able to backport this to the release/12.x as it didn't converted Xpriv to Xpub properly for multipath: https://github.com/rust-bitcoin/rust-miniscript/pull/757, and get this error.
